### DISCUSSION TITLE
Multiple patches, embedded on same page?

### DISCUSSION THREAD

**Discussion by timpulver** [cables-gl/cables_docs#595] [Answers & Questions] [answered]
Do you have some tips on how to embed multiple patches on the same website?

```html
<script type="text/javascript" src="/src/cables/patch1/js/libs.core.min.js"></script>
<script type="text/javascript" src="/src/cables/patch1/js/cables.min.js"></script>
<script type="text/javascript" src="/src/cables/patch1/js/ops.js"></script>

<script type="text/javascript"  src="/src/cables/patch1/js/cgl_shadermodifier.js"></script>
<script type="text/javascript"  src="/src/cables/patch1/js/cgl_copytexture.js"></script>
```

```javascript
const patch1 = new CABLES.Patch({
  patchFile: 'patch1/patch.json',
  glCanvasId: 'canvas1',
})

const patch2 = new CABLES.Patch({
  patchFile: 'patch2/patch.json',
  glCanvasId: 'canvas2',
})

const patch3 = new CABLES.Patch({
  patchFile: 'patch3/patch.json',
  glCanvasId: 'canvas3',
})
```

Which of the JavaScript files do I need to load _per patch_? Or asked differently: which of the JavaScript files do I only need to load once (from patch 1)?

For responsiveness, I would need to create a [ResizeObserver](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver) for each canvas, I guess.

**Comment by steam0r** [comment, marked as answer]
`libs.core.min.js`and `cables.min.js` can be shared, ops.js basically needs to be combined, or renamed to `ops1.js`, `ops2.js` and so on. to not have duplicate ops, we have that option `-C` in https://github.com/cables-gl/cables-cli, maybe give that a try.

window.addEventListener("resize", ...);
window.addEventListener("orientationchange", ...);

should be fine for "responsiveness".

**Reply by timpulver** [reply]
Thanks, this works well!
For others—the patches have to be fetched individually with the cli, and then another fetch for the combined ops code (with the `-C` option.
